### DISCUSSION TITLE
Autocompletion of class members

### DISCUSSION THREAD

**Discussion by iamreinder** [phpactor/phpactor#2061] [Q&A] [answered]
Hello all,

I've been looking into using Neovim in my workflow recently and love it so far. I've got quite a nice config going already and I use Phpactor as LSP for PHP projects. It's really good. There's just one thing I'd like to improve of which I'm not sure how to do it and if it's even possible. Maybe someone here can help me out.

I also work with TypeScript (with tsserver as LSP) and whenever I try to complete an object, it's class members are visible in the suggestions popup as soon as I type the period. If I try the same with a class in PHP, e.g. `$myObject->` it doesn't suggest anything untill I type another character, for instance `$myObject->g` for listing getter functions. I would like the autocompletion of PHP to work the same as JavaScript/TypeScript. That it provides suggestions as soon as you typed the arrow. Is that possible?

Thanks in advance.

/edit: Completing static members does work as I describe above, e.g. typing `$myObject::` immediately provides a list of suggestions. Also, if I type `$myObject->g` and than backspace to `$myObject->` I do get a list of suggestions of all members of that class.

**Comment by dantleech** [comment, marked as answer]
Hmm, this should already be the case. I don't this there is anything in Phpactor that would cause this limitation.

**Reply by iamreinder** [reply]
Hmpf, okay. I'll try to create a new barebones config with just Phpactor, see if it works, and than add features from there to see which plugin or config breaks the behaviour. Thanks for the quick response.

**Reply by iamreinder** [reply]
I found it! It was Treesitter autotag that caused the problem.
